### PR TITLE
Add prefer match file name

### PR DESCRIPTION
**Pull request by DenisBogatirov**

### CODE REVIEW

File: lint_test/prefer_match_file_name_test.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// ignore_for_file: unused_element, unused_field
+
+/// Check the `prefer-match-file-name` rule
+class _AnotherPrivateClass {}
+
+class PreferMatchFileNameTest {}
```
Comment (yurii-prykhodko-solid) [inline lint_test/prefer_match_file_name_test.dart:6]: Should be a lint here.
```suggestion
// expect_lint: prefer-match-file-name
class PreferMatchFileNameTest {}
```

Comment (DenisBogatirov) [reply lint_test/prefer_match_file_name_test.dart:7]: It matches file name. 
Original rule from DCM ignores private declarations and checks only first public declaration

Comment (yurii-prykhodko-solid) [reply lint_test/prefer_match_file_name_test.dart:7]: Ah, my bad.

File: lint_test/prefer_match_file_name_test.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// ignore_for_file: unused_element, unused_field
+
+/// Check the `prefer-match-file-name` rule
+class _AnotherPrivateClass {}
+
+// expect_lint: prefer-match-file-name
```
Comment (yurii-prykhodko-solid) [inline lint_test/prefer_match_file_name_test.dart:6]:
```suggestion
```

**Review by yurii-prykhodko-solid — APPROVED** [review]
LGTM

**Pull request merged**